pilot/pkg/model/service.go

```diff
@@ -70,8 +71,13 @@ type Service struct {
 	Hostname host.Name `json:"hostname"`
 
 	// Address specifies the service IPv4 address of the load balancer
+	// Do not access directly. Use GetServiceAddressForProxy
 	Address string `json:"address,omitempty"`
 
+	// AutoAllocatedAddress specifies the automatically allocated IPv4 address
+	// out of 240.240.0.0/16 subnet for service entries with non-wildcard hostnames,
+	AutoAllocatedAddress string `json:"autoAllocatedAddress,omitempty"`
```

Comment:
confused about this comment without looking implement details

Reply:
what is the confusion?

Reply:
How are the auto-allocated addresses synced ?

Reply:
I mean: since we have no RFC for this feature, at least the comments should include info about how it works.

Reply:
synced with what? this is per istiod. it does not have to be global as these addresses are not reachable outside.

Reply:
If you reconnect to another instance the entire xds config is going to be replaced and drain all connection then?

Reply:
And if you delete the oldest service then every other service also is changed?

Reply:
Good point. 
The current algorithm to allocate IPs is deterministic across all istiods. So, when there is no change, given two istiods with exact same set of services, there should be no change in XDS as the algorithm is just a dumb iterative one that allocates sequentially.

Remember, that the auto allocation is ONLY for service entries that do not have an IP address specified already. It is not for k8s. So, if one of these service entries without an IP is deleted, then yes, the IPs will shift and will cause listener drain for those listeners of TCP service entries and RDS drain on HTTP service entries (rds drain is not that problematic). But keep in mind that when a service entry is deleted (even if there were no auto allocation), there will be a RDS reload (if its a HTTP service). And if its a TCP service, there will be no impact. But if its a TCP service, odds are, without the DNS thing, the user is already having problems due to a wildcard listener on 0.0.0.0:3306 conflicting with others in the system. So essentially, we are exchanging one usability problem for another (tcp listener drain). That said, I think we can make the IP allocation lot more deterministic using some form of hashing. Will add a TODO here.